### ISSUE TITLE
CTR DRBG dependency on AES

### ISSUE THREAD

**Issue by smuellerDD** [usnistgov/ACVP#691]
In CAVS, the CTR DRBG de[pends on AES-ECB.

The ACVP spec for the CTR DRBG is not so clear as it only seems to refer to AES without the block chaining mode.

The CTR DRBG technically can be implemented with CTR-AES (see the Linux kernel DRBG where I implemented that). Is it permissible to satisfy the requirement from DRBG to AES by only testing the CTR AES or is it required to test AES-ECB?

**Comment by celic** [comment]
Wouldn't AES-CTR have its own dependency on AES-ECB? The chain of dependencies there would seem to imply that CTR-DRBG is dependent on both AES-CTR and AES-ECB.

**Comment by smuellerDD** [comment]
Why would CTR have a dependency on ECB? Other block chaining modes also do not have such dependency on ECB?! For CAVS the ECB dependency was there as there was no independent CTR test.

Ciao
Stephan

—
Cui bono?

**Comment by celic** [comment]
You are right. The ECB dependency for CTR was because the testing in the past did not exist. As we support both CAVS and ACVP, this could be a bit confusing. For CAVS, you must continue to exercise that dependency, but for ACVP as the testing has improved, it is no longer necessary. 

To go back to your original question, I'd say yes, it is permissible to test CTR as a prerequisite for CTR-DRBG, TDES or AES. The DRBG specification was probably written before the new CTR tests came about.

**Comment by smuellerDD** [comment]
Thank you for the clarification. Shall we close the issue or do you want to keep it open to note this issue somewhere?
